frontends/precinct-scanner/src/components/change_precinct_button.tsx

```diff
@@ -81,43 +141,21 @@ export function ChangePrecinctButton({
                 polls must be reopened. Please select a precinct and confirm
                 below.
               </p>
-              <Prose textCenter>
-                <Select
-                  id="selectPrecinct"
-                  data-testid="selectPrecinct"
-                  value={pendingPrecinctSelectionValue}
-                  onBlur={handlePendingPrecinctSelectionChange}
-                  onChange={handlePendingPrecinctSelectionChange}
-                >
-                  {election.precincts.length > 1 && (
-                    <option value={ALL_PRECINCTS_OPTION_VALUE}>
-                      {ALL_PRECINCTS_NAME}
-                    </option>
-                  )}
-                  {[...election.precincts]
-                    .sort((a, b) =>
-                      a.name.localeCompare(b.name, undefined, {
-                        ignorePunctuation: true,
-                      })
-                    )
-                    .map((precinct) => (
-                      <option key={precinct.id} value={precinct.id}>
-                        {precinct.name}
-                      </option>
-                    ))}
-                </Select>
-              </Prose>
+              <Prose textCenter>{precinctSelectDropdown}</Prose>
             </Prose>
           }
           actions={
             <React.Fragment>
               <Button
                 danger
                 onPress={confirmPrecinctChange}
-                disabled={areEqualPrecinctSelections(
-                  initialPrecinctSelection,
-                  pendingPrecinctSelection
-                )}
+                disabled={
+                  !unconfirmedPrecinctSelection ||
+                  areEqualPrecinctSelections(
+                    assertDefined(appPrecinctSelection),
```

Comment:
I added this `assertDefined` utility because I think it's valuable to be able to indicate, in code, that something should be defined, just like to do with `assert`. It is essentially re-introducing the bang operator, and `assert` is preferable where possible. But these cases, using the existence of a prop as a condition for rendering the element may create the false impression that the element is not meant to be rendered under those conditions.

In @jonahkagan's words, better to fail loudly. cc @arsalansufi who discussed this with me.

Reply:
I've written a utility like this at various times in my various TypeScript projects. It's better than a type cast/assertion (whether `as` or `!`) because it will fail _here_ and not at some later point where the value being used is not as expected. It's sometimes worse in terms of readability, but on the whole I think it's good.

Reply:
Sounds good. I ended up not needing it right here but have left it in this PR because I can anticipate needing it again.